File: src/main/java/com/indiepost/model/analytics/Campaign.java
package com.indiepost.model.analytics;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by jake on 8/7/17.
 */
@Entity
@Table(name = "Campaigns")
public class Campaign {

    @Id
    @GeneratedValue
    private Long id;

    @NotNull
    private String clientName;

    @NotNull
    private String name;

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "campaign", cascade = CascadeType.ALL)
    private List<Link> links = new ArrayList<>();

    @NotNull
    private LocalDateTime startAt;

    @NotNull
    private LocalDateTime endAt;

    @NotNull
    private LocalDateTime createdAt;

    private Long goal;

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getClientName() {
        return clientName;
    }

    public void setClientName(String clientName) {
        this.clientName = clientName;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Link> getLinks() {
        return links;
    }

    public void setLinks(List<Link> links) {
        this.links = links;
    }

    public LocalDateTime getStartAt() {
        return startAt;
    }

    public void setStartAt(LocalDateTime startAt) {
        this.startAt = startAt;
    }

    public LocalDateTime getEndAt() {
        return endAt;
    }

    public void setEndAt(LocalDateTime endAt) {
        this.endAt = endAt;
    }

    public Long getGoal() {
        return goal;
    }

    public void setGoal(Long goal) {
        this.goal = goal;
    }
}

File: src/main/kotlin/com/indiepost/model/analytics/Campaign.kt
package com.indiepost.model.analytics

import java.time.LocalDateTime
import java.util.*
import javax.persistence.*
import javax.validation.constraints.NotNull

/**
 * Created by jake on 8/7/17.
 */
@Entity
@Table(name = "Campaigns")
data class Campaign(
        @Id
        @GeneratedValue
        var id: Long? = null,

        @NotNull
        var clientName: String? = null,

        @NotNull
        var name: String? = null,

        @NotNull
        var startAt: LocalDateTime? = null,

        @NotNull
        var endAt: LocalDateTime? = null,

        @NotNull
        var createdAt: LocalDateTime? = null,

        var goal: Long? = null
) {
    @OneToMany(fetch = FetchType.LAZY, mappedBy = "campaign", cascade = [CascadeType.ALL])
    var links: MutableList<Link> = ArrayList()

}
